assignments/arrays.js

```diff
@@ -62,35 +69,59 @@ let inventory = [{"id":1,"car_make":"Lincoln","car_model":"Navigator","car_year"
-console.log(`Car 33 is a *car year goes here* *car make goes here* *car model goes here*` );
-
+// The dealer can't recall the information for a car with an id of 33 on his lot. 
+//Help the dealer find out which car has an id of 33 by logging the car's year, make, and model in 
+//the console log provided to you below:
+console.log(`Car 33 is a ${inventory[32].car_year} ${inventory[32].car_make} ${inventory[32].car_model}` );
 
 
 // ==== Challenge 2 ====
 // The dealer needs the information on the last car in their inventory.  What is the make and model of the last car in the inventory?  Log the make and model into the console.
-let lastCar = 0;
-console.log();
+let lastCar = inventory.length - 1;
+console.log(inventory[lastCar].car_make, inventory[lastCar].car_model);
 
 // ==== Challenge 3 ====
 // The marketing team wants the car models listed alphabetically on the website. Sort all the car model names into alphabetical order and log the results in the console
 let carModels = [];
-console.log();
+for (let i = 0; i < inventory.length; i++) {
+    carModels.push(inventory[i].car_model);
+} 
+console.log(carModels.sort());
 
 // ==== Challenge 4 ====
 // The accounting team needs all the years from every car on the lot. Create a new array from the dealer data containing only the car years and log the result in the console.
 let carYears = [];
-console.log();
+
+for(car in inventory){
+    carYears.push(inventory[car].car_year);
+}
+
+console.log(carYears);
 
 // ==== Challenge 5 ====
-// The car lot manager needs to find out how many cars are older than the year 2000. Using the carYears array you just created, find out how many cars were made before the year 2000 by populating the array oldCars and logging it's length.
+// The car lot manager needs to find out how many cars are older than the year 2000. Using the carYears array you just created, find out how many cars were made 
+//before the year 2000 by populating the array oldCars and logging it's length.
 let oldCars =[];
-console.log(); 
+for (let i = 0; i < carYears.length; i++) {
+    if (carYears[i] < 2000) {
+        oldCars.push(carYears[i]);
+    }                        
+}
+//carYear[0], carYear[50]
+console.log(oldCars); 
 
 // ==== Challenge 6 ====
-// A buyer is interested in seeing only BMW and Audi cars within the inventory.  Return an array that only contains BMW and Audi cars.  Once you have populated the BMWAndAudi array, use JSON.stringify() to show the results of the array in the console.
+// A buyer is interested in seeing only BMW and Audi cars within the inventory.  Return an array that only contains BMW and Audi cars.  Once you have populated the BMWAndAudi array, 
+//use JSON.stringify() to show the results of the array in the console.
 let BMWAndAudi =[];
-console.log();
+
+for(car in inventory){
+    if(inventory[car].car_make === 'BMW' || "Audi"){
```

Comment:
this evaluates as:
`if ( (inventory[car].car_make === 'BMW') || "Audi") {...`.

To javascript this reads as  
`if (a || b)` where a equals `(inventory[car].car_make === 'BMW')` and b equals `'Audi'`. Since it is a logical comparison, it will convert b to a Boolean, and any non-empty strings in JS always evaluate to true. So that means you're evaluating `if (a || true)`, which is always true.

What you want here is `if (inventory[car].car_make === 'BMW' || inventory[car].car_make === "Audi")`